### PR TITLE
Dnf shell tests

### PR DESCRIPTION
**Pull request by kkaarreell**
These 4 commits implement dnf shell tests for:
  * installing, updating and removing a package using dnf shell
  * running dnf shell commands from a batch file or stdin
  * switching conflicting packages using dnf shell
  * installing, updating and removing a package group using dnf shell
plus adding new step implementing dnf shell support

### CODE REVIEW

File: dnf-docker-test/features/shell-2.feature [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  @setup
+  Scenario: Preparing the test repository
+    Given repository "TestRepoA" with packages
+         | Package  | Tag       | Value |
+         | TestPkgA |           |       |
+
+  Scenario: Passing shell commands stored in a file
+    Given a file "/tmp/dnf_shell_transaction" with
+"""
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/shell-2.feature]: please indent such blocks

File: dnf-docker-test/features/steps/shell_commands.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+@then('the command output should contain "{text}"')
+def step_the_command_output_shoud_contain(context, text):
+    assert text in context.cmd_result
+
+
+@then('the command output should not contain "{text}"')
+def step_the_command_output_shoud_not_contain(context, text):
+    assert not (text in context.cmd_result)
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/shell_commands.py]: braces are not needed

Comment (kkaarreell) [inline dnf-docker-test/features/steps/shell_commands.py]: I will, though I find TestPkg much better once you start using also repositories, package groups etc.

**Comment by MichaelMraka** [comment]
@dnf-bot r+

**Pull request merged**